### PR TITLE
Show All Employees and Add an Employee

### PR DESCRIPTION
**Pull request by ManvendraSingh-Zemoso**

### CODE REVIEW

File: src/main/java/com/zemoso/demo/controller/EmployeeController.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    private EmployeeMapper employeeMapper;
+
+    @RequestMapping(method = RequestMethod.POST)
+    public ResponseEntity <Map<String,Map<String,String>>> addEmployee(@RequestBody Map<String,Map<String,String>> map){
+        Map<String,String> employee = map.get("employee");
+        Employee newEmployee = employeeService.saveEmployee(employeeMapper.mapToEmployee(employee));
+        Map<String,Map<String,String>> responseMap = new HashMap<>();
+        responseMap.put("employee",employeeMapper.employeeToMap(newEmployee));
+        return ResponseEntity.ok().body(responseMap);
+    }
```
Comment (shashanksheela) [inline src/main/java/com/zemoso/demo/controller/EmployeeController.java]: use httpstatus

File: src/main/java/com/zemoso/demo/controller/EmployeeController.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        updateEmployee.setLocation(employee.get(Constant.LOCATION));
+        updateEmployee.setJoinDate(employee.get(Constant.JOIN_DATE));
+        updateEmployee.setPicture(employee.get(Constant.PICTURE));
+        updateEmployee.setBio(employee.get(Constant.BIO));
+        updateEmployee.setSkills(employee.get(Constant.SKILLS));
+        updateEmployee.setDepartment(employee.get(Constant.DEPARTMENT));
+        updateEmployee.setProject(employee.get(Constant.PROJECT));
+        updateEmployee.setReportEmployee(employee.get(Constant.REPORT_EMPLOYEE));
+        employeeService.saveEmployee(updateEmployee);
+        return ResponseEntity.ok().body(employeeMapper.employeeToMap(updateEmployee));
```
Comment (shashanksheela) [inline src/main/java/com/zemoso/demo/controller/EmployeeController.java]: try to use beanutils set proprty method

File: src/main/java/com/zemoso/demo/controller/EmployeeController.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        updateEmployee.setEmailId(employee.get(Constant.EMAIL_ID));
+        updateEmployee.setPhoneNo(employee.get(Constant.PHONE_NO));
+        updateEmployee.setRole(employee.get(Constant.ROLE));
+        updateEmployee.setLocation(employee.get(Constant.LOCATION));
+        updateEmployee.setJoinDate(employee.get(Constant.JOIN_DATE));
+        updateEmployee.setPicture(employee.get(Constant.PICTURE));
+        updateEmployee.setBio(employee.get(Constant.BIO));
+        updateEmployee.setSkills(employee.get(Constant.SKILLS));
+        updateEmployee.setDepartment(employee.get(Constant.DEPARTMENT));
+        updateEmployee.setProject(employee.get(Constant.PROJECT));
```
Comment (shashanksheela) [inline src/main/java/com/zemoso/demo/controller/EmployeeController.java]: as discussed move this part to a method

File: src/main/java/com/zemoso/demo/model/Employee.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import javax.persistence.Id;
+import javax.validation.constraints.NotNull;
+
+@Data
+@Entity
+public class Employee {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
```
Comment (shashanksheela) [inline src/main/java/com/zemoso/demo/model/Employee.java]: use GenerationType.Sequence

File: src/main/java/com/zemoso/demo/controller/EmployeeController.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    private ResponseEntity <Map<String,List<Map<String,String>>>> getAllEmployees(){
+        List<Employee> employeeList = employeeService.getAllEmployees();
+        List<Map<String,String>> response = new ArrayList<>();
+        employeeList.forEach(item ->{
+            Map<String,String> employee = employeeMapper.employeeToMap(item);
+            response.add(employee);
+        });
```
Comment (shashanksheela) [inline src/main/java/com/zemoso/demo/controller/EmployeeController.java]: write comments everywhere

File: src/main/java/com/zemoso/demo/service/EmployeeDepartmentServiceImplementation.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import java.util.List;
+
+@Service
+public class EmployeeDepartmentServiceImplementation implements EmployeeDepartmentService {
+
+    @Autowired
+    private EmployeeDepartmentRepository employeeDepartmentRepository;
+
+    public List<EmployeeDepartment> getAllEmployeeDepartments(){ return (List<EmployeeDepartment>)employeeDepartmentRepository.findAll();}
+}
```
Comment (shashanksheela) [inline src/main/java/com/zemoso/demo/service/EmployeeDepartmentServiceImplementation.java]: remove employee from names

File: src/main/java/com/zemoso/demo/model/EmployeeSkill.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+public class EmployeeSkill {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @NotNull
+    private String name;
+    private String description;
+}
```
Comment (shashanksheela) [inline src/main/java/com/zemoso/demo/model/EmployeeSkill.java]: use base entity

File: frontend/app/controllers/employees/new.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
+import Controller from '@ember/controller';
+import { filter } from '@ember/object/computed';
+import { computed } from '@ember/object';
+import { A } from '@ember/array';
+
+export default Controller.extend({
+  roles: A(['Engineer','Developer','Manager','Director']),
```
Comment (vijaysai88zemoso) [inline frontend/app/controllers/employees/new.js]: Fetch it from the backend

File: frontend/app/routes/employees/new.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import Route from '@ember/routing/route';
+import RSVP from 'rsvp';
+
+export default Route.extend({
+  model() {
+    return RSVP.hash({
+      employee: this.get('store').createRecord('employee'),
+      reportToEmployees: this.get('store').findAll('employee')
```
Comment (vijaysai88zemoso) [inline frontend/app/routes/employees/new.js]: Make the separate request for employee names

File: frontend/app/templates/employees/index.hbs [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+          <p>Employees</p>
+        </div>
+        <div>
+          {{#paper-button class="primary-button"raised=true primary=true onClick=(action "addEmployee")}}+Add Employee{{/paper-button}}
+        </div>
+      </div>
+      {{#each model as |employee|}}
+        <div class="flex-20 cards">
+          <div class="layout-column  flex">
+            {{#paper-card class="card" as |card|}}
```
Comment (vijaysai88zemoso) [inline frontend/app/templates/employees/index.hbs]: Put this in separate component

**Pull request closed without merging**